Snakeyaml NoClassDefFoundError when using config4k configuration, @MicronautTest and propertySources

### Expected Behavior

When writing a Micronaut service using kotlin and config4k configuration (no `yml` files in the service at all), I would expect to be able to load a `.conf` file in my tests using the `propertySources` attribute of the `@MicronautTest` annotation just like I can when using `.properties` or `.yml` confilguration:

```kotlin
@MicronautTest(propertySources = ["classpath:my-greeting.conf"])
class GreetingControllerWithSpecifiedConfFileTest(private val greetingConfig: GreetingConfig) {

  @Inject @Client("/") lateinit var client: HttpClient

  @Test
  internal fun `the greeting controller returns the configured message`() {
    val response = client.toBlocking().retrieve("/greeting/hello")
    println("Greeting should be '${greetingConfig.message}'")
    assertEquals(greetingConfig.message, response)
  }
}
```


### Actual Behaviour

Now Micronaut 4.x does not expose SnakeYAML as a transitive dependency I am seeing Snakeyaml NoClassDefFoundError errors when I try to specify a specific `.conf` file in my tests using the `propertySources` attribute of the `@MicronautTest` annotation:

```
Error loading service [io.micronaut.context.env.yaml.YamlPropertySourceLoader]: org/yaml/snakeyaml/constructor/BaseConstructor
java.util.ServiceConfigurationError: Error loading service [io.micronaut.context.env.yaml.YamlPropertySourceLoader]: org/yaml/snakeyaml/constructor/BaseConstructor
	at io.micronaut.core.io.service.DefaultServiceDefinition.lambda$load$0(DefaultServiceDefinition.java:76)
	at java.base/java.util.Optional.map(Optional.java:260)
	at io.micronaut.core.io.service.DefaultServiceDefinition.load(DefaultServiceDefinition.java:72)
	at io.micronaut.test.extensions.AbstractMicronautExtension.readPropertySourceLoaderMap(AbstractMicronautExtension.java:520)
	at io.micronaut.test.extensions.AbstractMicronautExtension.beforeClass(AbstractMicronautExtension.java:269)
	at io.micronaut.test.extensions.junit5.MicronautJunit5Extension.beforeAll(MicronautJunit5Extension.java:84)
	at ..........
Caused by: java.lang.NoClassDefFoundError: org/yaml/snakeyaml/constructor/BaseConstructor
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3373)
	at java.base/java.lang.Class.getConstructor0(Class.java:3578)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2754)
	at io.micronaut.core.io.service.DefaultServiceDefinition.lambda$load$0(DefaultServiceDefinition.java:74)
	... 62 more
```

Specifying the environment using `@MicronautTest(environments = ["mytest"])` seems to work fine with `.conf` files as does specifying the config value using the `@Property` attribute:

```kotlin
@MicronautTest
@Property(name = "greeting.message", value = "Hello from the custom property defined in the test")
```

### Steps To Reproduce

1. Pull down the [example repository](https://github.com/leeturner/micronaut-test-config4k)
2. Open the project inside an IDE of your choice and run all the tests, or
3. Run `./gradlew clean test`
4. All tests pass  except `GreetingControllerWithSpecifiedConfFileTest` which fails with:

```
GreetingControllerWithSpecifiedConfFileTest > initializationError FAILED
    java.util.ServiceConfigurationError at DefaultServiceDefinition.java:76
        Caused by: java.lang.NoClassDefFoundError at Class.java:-2
            Caused by: java.lang.ClassNotFoundException at BuiltinClassLoader.java:641
```

### Environment Information

- Mac OS 12.5 (21G72)
- Java version - `openjdk 17.0.3 2022-04-19`:
```
OpenJDK Runtime Environment GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06)
OpenJDK 64-Bit Server VM GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06, mixed mode, sharing) 
```

### Example Application

https://github.com/leeturner/micronaut-test-config4k

### Version

4.0.0-M4

--- a/inject/src/main/java/io/micronaut/context/env/yaml/YamlPropertySourceLoader.java
+++ b/inject/src/main/java/io/micronaut/context/env/yaml/YamlPropertySourceLoader.java
@@ -56,8 +56,7 @@ public class YamlPropertySourceLoader extends AbstractPropertySourceLoader {
             System.setProperty("java.runtime.name", "Unknown");
         }
 
-        Yaml yaml = new Yaml(new CustomSafeConstructor());
-        Iterable<Object> objects = yaml.loadAll(input);
+        Iterable<Object> objects = Wrapper.loadObjects(input);
         Iterator<Object> i = objects.iterator();
         if (i.hasNext()) {
             while (i.hasNext()) {
@@ -83,4 +82,12 @@ public class YamlPropertySourceLoader extends AbstractPropertySourceLoader {
         }
     }
 
+    private static class Wrapper {
+        // in nested class to prevent NCDFE
+
+        private static Iterable<Object> loadObjects(InputStream input) {
+            Yaml yaml = new Yaml(new CustomSafeConstructor());
+            return yaml.loadAll(input);
+        }
+    }
 }

